test/api/task.test.ts

```diff
@@ -0,0 +1,181 @@
+import { createHandler, type ServeHandlerInfo } from '$fresh/server.ts';
+import manifest from '../../fresh.gen.ts';
+import config from '../../fresh.config.ts';
+import { assertEquals } from '$std/assert/mod.ts';
+
+const hostname = '127.0.0.1';
+const root = `http://${hostname}`;
+
+const CONN_INFO: ServeHandlerInfo = {
+  remoteAddr: { hostname, port: 53496, transport: 'tcp' },
+};
+
+await Deno.test('Tasks', async (t) => {
+  const handler = await createHandler(manifest, config);
+  const name = crypto.randomUUID();
+
+  let resp: Response | undefined = undefined;
+
+  await handler(
+    new Request(`${root}/api/users`, {
+      method: 'POST',
+      body: JSON.stringify({
+        name,
+        password: 'test',
+      }),
+    }),
+    CONN_INFO,
+  );
+
+  resp = await handler(
+    new Request(`${root}/api/sessions`, {
+      method: 'POST',
+      body: JSON.stringify({
+        name,
+        password: 'test',
+      }),
+    }),
+    CONN_INFO,
+  );
+
+  const { _UserId, token } = await resp!.json();
+  const headers = {
+    'Authorization': `bearer ${token}`,
+  };
+
+  await t.step('create', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'POST',
+        body: '',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'POST',
+        body: '',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 500);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify({
+          name: 'Wash the cats',
+        }),
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+
+  const task = await resp!.json();
+
+  await t.step('get tasks', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'GET',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'GET',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+
+  await t.step('single task', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'GET',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks/${crypto.randomUUID()}`, {
```

Comment:
Maybe use Task.id(), once you've converted Task to be based on Keyed.

Reply:
I don't understand this sentence. Particularly this bit 'once you've converted Task to be based on Keyed'. How is task not based on Keyed here if Keyed is necessary to create a new Task?

Reply:
Sorry, I didn't notice that Task got updated.  strike that part.   `Task.id()` will give you a ULID that's in the right format for this request.  I like there *also* being a test where the task ID is not syntactically correct.